### PR TITLE
change interfaces to comply with paginated search results

### PR DESCRIPTION
**Pull request by Qrzy**
Search results in `useProduct` and `useUserAddress` can be paginated, hence this change that follows the way the `useUserOrders` has been prepared for pagination.

### CODE REVIEW

File: packages/core/interfaces/src/index.ts [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { Ref } from '@vue/composition-api';
 type ComputedProperty<T> = Readonly<Ref<Readonly<T>>>;
 
 export interface UseProduct<PRODUCT> {
-  products: ComputedProperty<PRODUCT[]>;
-  totalProducts: ComputedProperty<number>;
+  products: {
```
Comment (andrzejewsky) [inline packages/core/interfaces/src/index.ts]: please use `products` and `totalProducts` instead

File: packages/core/interfaces/src/index.ts [hunk 2]
```diff
@@ -56,30 +58,27 @@ export interface UseUserOrders<ORDER> {
...
     [x: string]: any;
   }) => Promise<void>;
-  loading: Ref<boolean>;
+  loading: ComputedProperty<boolean>;
 }
 
-/** check if we always have those addresses together or we need pagination  */
 export interface UseUserAddress<ADDRESS> {
-  addresses: ComputedProperty<ADDRESS[]>;
+  addresses: {
```
Comment (andrzejewsky) [inline packages/core/interfaces/src/index.ts]: same here `addressess` and` totalAddressess`

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: packages/core/factories/src/useProductFactory.ts [hunk 3]
```diff
@@ -9,8 +9,13 @@ type SearchParams = {
   filters?: any;
 }
 
+export type ProductsSearchResult<PRODUCT> = {
+  data: PRODUCT[];
+  total: number;
+};
+
```
Comment (patzick) [inline packages/core/factories/src/useProductFactory.ts]:
```suggestion
export type SearchResult<T> = {
  data: T[];
  total: number;
};

```

**Review by patzick — CHANGES_REQUESTED** [review]
Just one note + point from @andrzejewsky

File: packages/core/factories/src/useProductFactory.ts (hunk 3 above)
Comment (patzick) [reply packages/core/factories/src/useProductFactory.ts]: this can be used for all searches

File: packages/core/factories/src/index.ts [hunk 4]
```diff
@@ -1,3 +1,7 @@
+import Vue from 'vue';
+import CompositionApi from '@vue/composition-api';
+Vue.use(CompositionApi);
+
```
Comment (patzick) [inline packages/core/factories/src/index.ts]:
```suggestion
```

this shouldn't be here

**Review by patzick — CHANGES_REQUESTED** [review]
Just one thing from me :)

**Review by patzick — APPROVED** [review]
👍

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**